src/Stack/Build/Haddock.hs

```diff
@@ -114,6 +115,10 @@ shouldHaddockPackage bopts wanted name =
 shouldHaddockDeps :: BuildOpts -> Bool
 shouldHaddockDeps bopts = fromMaybe (boptsHaddock bopts) (boptsHaddockDeps bopts)
 
+-- | Determine whether to build haddocks for non-exported modules, like @cabal haddock --intenral@.
```

Comment:
Typo here in the flag.  I'd prefer to just not have this function and instead use `boptsHaddockInternal`.